src/main/java/edu/harvard/iq/dataverse/api/Users.java

```diff
@@ -266,4 +270,24 @@ public Response getTracesElement(@Context ContainerRequestContext crc, @Context
         }
     }
 
+    @POST
+    @Path("register")
+    public Response registerOidcUser(String body) {
+        if (!FeatureFlags.API_BEARER_AUTH.enabled()) {
+            return error(Response.Status.INTERNAL_SERVER_ERROR, BundleUtil.getStringFromBundle("users.api.errors.bearerAuthFeatureFlagDisabled"));
+        }
+        Optional<String> bearerToken = extractBearerTokenFromHeaderParam(httpRequest.getHeader(HttpHeaders.AUTHORIZATION));
+        if (bearerToken.isEmpty()) {
+            return error(Response.Status.BAD_REQUEST, BundleUtil.getStringFromBundle("users.api.errors.bearerTokenRequired"));
+        }
+        try {
+            JsonObject userJson = JsonUtil.getJsonObject(body);
```

Comment:
Can the json be optional/only used if needed? It seems odd (possibly dangerous?) to let someone use a token with one name/email and register an account for a different name/email. Further on in the code, the token is being used to call the UserInfo endpoint, so everything the provider shares is available.

Reply:
@qqmyers Are you suggesting that we send optional "preferred" fields, such as preferredEmailAddress and preferredUsername, and only use them when the IdP is unable to provide us with this information?

I’m concerned that a user might believe they registered with a specific email and username, while different values were actually saved. However, this issue could be avoided if the SPA retrieves these values from the IdP or from JWT token claims (if available) and displays them in the new user registration form as uneditable fields.

Reply:
I was thinking more that sending a name/email that is inconsistent with the OIDC account would fail, and that you wouldn't need to send JSON at all if the OIDC account includes all needed fields / only send the fields missing from the OIDC account.